### DISCUSSION TITLE
How is the git sha that triggers a release tied to a release number semantic-release generates?  What to do if malicious actor deletes and recreates and old git tag pointing to a new (bad) git sha?

### DISCUSSION THREAD

**Discussion by TCModus** [semantic-release/semantic-release#2899] [Q&A] [answered]
How is the git sha that triggers a release tied to a release number semantic-release generates?  

What to do if malicious actor deletes and recreates and old git tag pointing to a new (bad) git sha?

For example: if semantic-release generates version 1.2.3 (creating the git tag, the github release, and the npm dist-tag) that points to the sha XYZ in git, then later on a malicious actor deletes the git tag, release, and dist-tag and then creates a malicious change in the source code and generates a new tag for 1.2.3 pointing to the bad code, would there be a way to tell from the artifacts originally created when 1.2.3 was made that the sha doesn't match what originally pointed to 1.2.3?

**Comment by seebeen** [comment, marked as answer]
Turn on GPG signing, and you've solved the problem.
